### ISSUE TITLE
Wrong version of networkx for python 3.7

### ISSUE THREAD

**Issue by TylerAnderson22** [conda-forge/networkx-feedstock#44]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

In an empty conda envrionment I run `conda install python=3.7 networkx`.

I end up getting networkx version `2.7.1`.

When I go to the github tag for networkx version `2.7.1` I see that there is a requirement for python >=3.8: https://github.com/networkx/networkx/blob/networkx-2.7.1/setup.py#L195

Looking at commits on networkx-feedstock I see the first commit for 2.7.1 [here](https://github.com/conda-forge/networkx-feedstock/commit/b871075db44054acabcaba96aa046e99d7c39733) and then a correction for the version [here](https://github.com/conda-forge/networkx-feedstock/commit/89d2c0221d2e709c78d0cffc18d50859f1b4b8ff). The correction fixes the dependency issue.

I am getting the first commit, build 0 of networkx 2.7.1. This is unexpected behavior. Is this avoidable on my end or is this something that can be fixed on your end?

### Installed packages

```shell
# packages in environment at /home/tanderson/.conda/envs/networkx.test:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
brotli                    1.0.9                h166bdaf_8    conda-forge
brotli-bin                1.0.9                h166bdaf_8    conda-forge
ca-certificates           2022.9.24            ha878542_0    conda-forge
certifi                   2022.9.24          pyhd8ed1ab_0    conda-forge
cycler                    0.11.0             pyhd8ed1ab_0    conda-forge
fonttools                 4.38.0           py37h540881e_0    conda-forge
freetype                  2.12.1               hca18f0e_0    conda-forge
jpeg                      9e                   h166bdaf_2    conda-forge
kiwisolver                1.4.4            py37h7cecad7_0    conda-forge
lcms2                     2.14                 h6ed2654_0    conda-forge
ld_impl_linux-64          2.39                 hc81fddc_0    conda-forge
lerc                      4.0.0                h27087fc_0    conda-forge
libblas                   3.9.0           16_linux64_openblas    conda-forge
libbrotlicommon           1.0.9                h166bdaf_8    conda-forge
libbrotlidec              1.0.9                h166bdaf_8    conda-forge
libbrotlienc              1.0.9                h166bdaf_8    conda-forge
libcblas                  3.9.0           16_linux64_openblas    conda-forge
libdeflate                1.14                 h166bdaf_0    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 12.2.0              h65d4601_19    conda-forge
libgfortran-ng            12.2.0              h69a702a_19    conda-forge
libgfortran5              12.2.0              h337968e_19    conda-forge
libgomp                   12.2.0              h65d4601_19    conda-forge
liblapack                 3.9.0           16_linux64_openblas    conda-forge
libnsl                    2.0.0                h7f98852_0    conda-forge
libopenblas               0.3.21          pthreads_h78a6416_3    conda-forge
libpng                    1.6.38               h753d276_0    conda-forge
libsqlite                 3.40.0               h753d276_0    conda-forge
libstdcxx-ng              12.2.0              h46fd767_19    conda-forge
libtiff                   4.4.0                h55922b4_4    conda-forge
libwebp-base              1.2.4                h166bdaf_0    conda-forge
libxcb                    1.13              h7f98852_1004    conda-forge
libzlib                   1.2.13               h166bdaf_4    conda-forge
matplotlib-base           3.5.3            py37hf395dca_2    conda-forge
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
ncurses                   6.3                  h27087fc_1    conda-forge
networkx                  2.7.1              pyhd8ed1ab_0    conda-forge
numpy                     1.21.6           py37h976b520_0    conda-forge
openjpeg                  2.5.0                h7d73246_1    conda-forge
openssl                   3.0.7                h166bdaf_0    conda-forge
packaging                 21.3               pyhd8ed1ab_0    conda-forge
pandas                    1.3.5            py37he8f5f7f_0    conda-forge
pillow                    9.2.0            py37h850a105_2    conda-forge
pip                       22.3.1             pyhd8ed1ab_0    conda-forge
pthread-stubs             0.4               h36c2ea0_1001    conda-forge
pyparsing                 3.0.9              pyhd8ed1ab_0    conda-forge
python                    3.7.12          hf930737_100_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python_abi                3.7                     2_cp37m    conda-forge
pytz                      2022.6             pyhd8ed1ab_0    conda-forge
readline                  8.1.2                h0f457ee_0    conda-forge
scipy                     1.7.3            py37hf2a6cf1_0    conda-forge
setuptools                59.8.0           py37h89c1867_1    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
sqlite                    3.40.0               h4ff8645_0    conda-forge
tk                        8.6.12               h27826a3_0    conda-forge
typing-extensions         4.4.0                hd8ed1ab_0    conda-forge
typing_extensions         4.4.0              pyha770c72_0    conda-forge
unicodedata2              14.0.0           py37h540881e_1    conda-forge
wheel                     0.38.4             pyhd8ed1ab_0    conda-forge
xorg-libxau               1.0.9                h7f98852_0    conda-forge
xorg-libxdmcp             1.1.3                h7f98852_0    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
zstd                      1.5.2                h6239696_4    conda-forge
```

### Environment info

```shell
active environment : networkx.test
    active env location : /home/tanderson/.conda/envs/networkx.test
            shell level : 2
       user config file : /home/tanderson/.condarc
 populated config files : /usr/share/conda/condarc.d/defaults.yaml
                          /home/tanderson/.condarc
          conda version : 4.6.14
    conda-build version : not installed
         python version : 3.6.8.final.0
       base environment : /usr  (read only)
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /var/cache/conda/pkgs
                          /home/tanderson/.conda/pkgs
       envs directories : /home/tanderson/.conda/envs
                          /usr/envs
               platform : linux-64
             user-agent : conda/4.6.14 requests/2.14.2 CPython/3.6.8 Linux/3.10.0-1160.80.1.el7.x86_64 centos/7.9.2009 glibc/2.17
                UID:GID : 123400014:123400014
             netrc file : None
           offline mode : False
```

**Comment by ocefpaf** [comment]
We should mark that package as broken. Do you want to send a PR to do that?


https://github.com/conda-forge/admin-requests#mark-packages-as-broken-on-conda-forge

**Comment by TylerAnderson22** [comment]
I will create the PR

**Comment by jakirkham** [comment]
Wonder if we can patch the repodata in this case, since it appears to be an incorrect requirement as opposed to an issue with the package itself

**Comment by four43** [comment]
Now that this is broken, it's no longer installable and throwing exceptions for us. Please don't "unpublish" versions

```
$ conda env create -f ./env-versions/v1/environment.lock.yml 
Collecting package metadata (repodata.json): done
Solving environment: failed

ResolvePackageNotFound: 
  - networkx==2.7.1=pyhd8ed1ab_0
  ```

**Comment by jakirkham** [comment]
FWIW think we are open to patching the repodata instead ( https://github.com/conda-forge/admin-requests/pull/525#issuecomment-1343158954 ) if someone wants to take that on

**Comment by MridulS** [comment]
@four43 https://github.com/conda-forge/admin-requests/pull/531 should fix the issue of networkx==2.7.1=pyhd8ed1ab_0 marked broken. The env should resolve.

@TylerAnderson22 https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/373 should fix the issue when someones runs `conda install python=3.7 networkx` now. It should install `networkx-2.6.3` in that env.
